pages/studentAssessmentInstanceExam/studentAssessmentInstanceExam.sql

```diff
@@ -12,7 +12,10 @@ SELECT
     (z.max_points IS NOT NULL) AS zone_has_max_points,
     z.best_questions AS zone_best_questions,
     (z.best_questions IS NOT NULL) AS zone_has_best_questions,
-    (SELECT count(*) FROM files AS f WHERE f.instance_question_id = iq.id AND f.deleted_at IS NULL) AS file_count
+    (SELECT count(*) FROM files AS f WHERE f.instance_question_id = iq.id AND f.deleted_at IS NULL) AS file_count,
+    qo.sequence_locked AS sequence_locked,
+    (lag(aq.effective_min_advance_perc) OVER w) AS prev_iq_min_advance_perc,
+    'Question ' || (lag(qo.question_number) OVER w) AS prev_iq_title
```

Comment:
What happens if we have three questions:

Question 1: `effective_min_advance_perc = 80`
Question 2: `effective_min_advance_perc = 0`
Question 3: `effective_min_advance_perc = 0`

In this case (assuming we are just starting the assessment), I think Q2 will display a lock icon and say "You need 80% on Q1", which is great.

But then Q3 will also display a lock icon (which is good) but it will say "You need 0% on Q2", which seems really weird. I don't think students will understand this.

Maybe it would be clearer to have the _first_ locked question say "You need 80% on Q1", but then all later locked questions should just say "You need to unlock the previous question before accessing this one" or something like that? I feel that this might be clearer for the student? This will hide the exact advance_perc for later questions, but I don't think that is very useful information anyway while the earlier questions are still locked.

An alternative to this would be to look backwards in the list of questions and find the closest previous question with `advance_perc > 0` and then report this. In the above example, this means that Q3 would also have a message like "You need 80% on Q1" (the same message as Q2). This is because Q2 and Q3 will both unlock as soon as the score on Q1 is over 80%.

Reply:
I can definitely see where you're coming from, unlocking multiple questions after an earlier question is answered is probably a good feature to have. That's not what the current version of this code does, though, so I think we should check in with @mfsilva22 to see if that's in the direction that she's looking for

Reply:
Done in recent commit